reworking the doc to make it more focused. Moving some parts to exten…

sridharangirish91 opened the pull request:
…ded section in the appendix

sridharangirish91 committed:
reworking the doc to make it more focused. Moving some parts to extended section in the appendix

sridharangirish91 committed:
rewording host to more specific terminology

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -99,35 +99,24 @@ informative:
 Wireless networks experience significant but transient variations in link quality that affect user experience.
 
-Collaborative signaling from host-to-network and server-to-network can improve the user experience by informing the network about the nature and relative importance of packets (frames, streams, etc.) without having to disclose the content of the packets. Moreover, the collaborative signalling may be enabled so that hosts are aware of the network's treatment of incoming packets. Also, host-to-netw ...
+Collaborative signaling from client-to-network and server-to-network can improve the user experience by informing the network about the nature and relative importance of packets (frames, streams, etc.) without having to disclose the content of the packets. Moreover, the collaborative signalling may be enabled so that the client and server are aware of the network's treatment of incoming packets.  ...
```
```suggestion
Collaborative signaling from client-to-network and server-to-network can improve the user experience by informing the network about the nature and relative importance of packets (frames, streams, etc.) without having to disclose the content of the packets. Moreover, the collaborative signaling may be enabled so that clients and servers are aware of the network's treatment of incoming packets. Also, client-to-network collaboration can be put in place without revealing the identity of the remote servers. This collaboration allows for differentiated services at the network (e.g., packet discard preference), the sender (e.g., adaptive transmission), or through cooperation of server/client and the network.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -99,35 +99,24 @@ informative:
+Collaborative signaling from client-to-network and server-to-network can improve the user experience by informing the network about the nature and relative importance of packets (frames, streams, etc.) without having to disclose the content of the packets. Moreover, the collaborative signalling may be enabled so that the client and server are aware of the network's treatment of incoming packets.  ...
 
-This document lists some use cases that illustrate the need for a mechanism to share metadata and outlines requirements for both host-to-network and server-to-network. The document focuses on intra-flow or flows bound to the same user.
+This document lists some use cases that illustrate the need for a mechanism to share metadata and outlines requirements for both client-to-network and server-to-network. The document focuses on intra-flow or flows bound to the same user.
```
```suggestion
This document lists some use cases that illustrate the need for a mechanism to share metadata and outlines requirements for both client-to-network and server-to-network. The document focuses on flows bound to the same user.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -99,35 +99,24 @@ informative:
-Users who are serviced via these networks use hosts which run various application clients; each having different connectivity needs for an optimal user experience.
-These needs are not frozen but change over time depending on the application and even depending on how an application is used (e.g., user's preferences).
-An explicit signal to the host can help to manage the use of available bandwidth better and better share it with requesting applications.
+Wireless networks including 5G and WLAN inherently experience large variations in link quality over sub-RTT intervals and on the other hand applications such as interactive media demand both low latency and high bandwidth. Maximizing network utilization and end user experience under such conditions is challenging. Factors that affect wireless networks include change in channel conditions, interfe ...
```
```suggestion
Wireless networks including 5G and WLAN inherently experience large variations in link quality over sub-RTT intervals and on the other hand applications such as interactive media demand both low latency and high bandwidth. Maximizing network utilization and end user experience under such conditions is challenging. Factors that affect wireless networks include change in channel conditions, interference between proximate cells, and end user mobility. These variations in link quality can be in the order of a millisecond or less {{5G-Lumos}} while congestion control takes several tens of milliseconds (more than one round-trip time (RTT)) to estimate data rate. Similarly, application servers that encode and serve live or interactive media take time to adjust the encoding level and other processes to match the network rate. End-to-end congestion control algorithms are far from being optimal when the link quality is highly variable in sub-RTT timeframes and the application demands both low latency and high bandwidth (e.g., {{Section 2.1 of ?RFC6077}}). In these conditions, applications settle for a lower throughput when latency is prioritized, or for higher throughput at the expense of much higher delays.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -99,35 +99,24 @@ informative:
-An explicit signal to the host can help to manage the use of available bandwidth better and better share it with requesting applications.
+Wireless networks including 5G and WLAN inherently experience large variations in link quality over sub-RTT intervals and on the other hand applications such as interactive media demand both low latency and high bandwidth. Maximizing network utilization and end user experience under such conditions is challenging. Factors that affect wireless networks include change in channel conditions, interfe ...
+
+While rate control based on feedback for a flow (UDP 4-tuple) is evidently not able adapt for sub-RTT changes in available wireless channel resources, the application server can provide information on a per-packet basis that a network shaper may use to utilize the available resources more effectively. {{5G-Octopus}} has shown for volumetric video packets and a rate controller that errs on the sid ...
```
```suggestion
While rate control based on feedback for a flow (UDP 4-tuple) is evidently not able adapt for sub-RTT changes in available wireless channel resources, an application server can provide information on a per-packet basis that a network shaper may use to allocate the available resources more effectively. {{5G-Octopus}} has shown for volumetric video packets and a rate controller that errs on the side of overestimation, that the network shaper can drop low priority frames of a group of pictures (corresponding to transient wireless link bandwidth drops) and still achieve significantly better performance than state-of-the-art based on feedback. With not fully encrypted packets, networks may use heuristics to build an "implicit signal" gleaned from a packet to prioritize or otherwise shape flows. However, implicit signals are not desirable as they lead to ossification of protocols as result of introducing unintended dependencies {{?RFC9419}}. When packet contents are encrypted, the approach of using implicit signals is no longer viable.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```suggestion
The requirements for this server-to-network (S2N) metadata are described in {{server-network}}. The client may provide information to an (access) router to drop lower priority marked packets of a flow (UDP 4-tuple) temporarily which can in turn redirect bandwidth to other flows of that network attachment, especially during a reactive event.
```

John-Kaippallimalil committed:
Update draft-kwbdgrr-tsvwg-net-collab-rqmts.md

Revised section 5.2 to add that for REQ-PACKET-PRIORITY the behavior is to drop packets of lower priority (i.e., they are near real-time and cannot tolerate delay)
Added 5.2.2 with REQ-PACKET-DELAY for packets that can tolerate some delay as in LE PHB RFC 8622.

John-Kaippallimalil reviewed (approved):
Revised 5.2 based on discussions yesterday. 
The revisions here are a good baseline for implementing the other aspects (the required set of requirements and the desired set of requirements. And revision of Operational to System Considerations).

sridharangirish91 committed:
Apply suggestions from code review

Co-authored-by: Med <mohamed.boucadair@orange.com>

sridharangirish91 committed:
nit

sridharangirish91 committed:
IncorrectConflictResolution correction

sridharangirish91 committed:
fix build break
